pkg/connector/source.go

```diff
@@ -106,13 +106,16 @@ func (s *source) Errors() <-chan error {
 	return s.errs
 }
 
-func (s *source) Validate(ctx context.Context, settings map[string]string) error {
+func (s *source) Validate(ctx context.Context, settings map[string]string) (err error) {
 	src, err := s.pluginDispenser.DispenseSource()
 	if err != nil {
 		return err
 	}
 	defer func() {
-		_ = src.Teardown(ctx)
+		tmpErr := src.Teardown(ctx)
+		err = cerrors.LogOrReplace(err, tmpErr, func() {
```

Comment:
Not really, it could be that `src.Configure` will fail in line 121. Notice that the function uses a named return variable, so `err` will contain the returned value and we are able to overwrite it in a deferred function.

Reply:
Named return variables :-1: :D